File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Myanmar.m
//
//  UIImage+Myanmar.m
//  CountriesFlags
//

#import "Countries/UIImage+Myanmar.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Myanmar)

+ (UIImage * _Nullable)flagMyanmarWithSize:(CGSize)size {
    UIColor* colorGreen = [UIColor colorWithRed:52.0f/255.0f
                                          green:178.0f/255.0f
                                           blue:51.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:234.0f/255.0f
                                        green:40.0f/255.0f
                                         blue:57.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:254.0f/255.0f
                                           green:203.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f * 2.0f, size.width, size.height / 3.0f));
    
    [colorWhite setFill];
    [[UIBezierPath starAtPoint:CGPointMake(size.width / 2.0, (size.height + (size.height / 6.0f * 0.422291236)) / 2.0f)
                        radius:size.height / 12.0f * 8.844582472 / 2.0f
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Myanmar.swift
//
//  UIImage+Myanmar.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Myanmar "wikipedia"
     
     Generates the flag of Myanmar.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Myanmar.
     */
    internal static func flagMyanmar(with size: CGSize) -> UIImage {
        let colorGreen = UIColor(red: 52.0 / 255.0,
                                 green: 178.0 / 255.0,
                                 blue: 51.0 / 255.0,
                                 alpha: 1.0)
        let colorRed = UIColor(red: 234.0 / 255.0,
                               green: 40.0 / 255.0,
                               blue: 57.0 / 255.0,
                               alpha: 1.0)
        let colorYellow = UIColor(red: 254.0 / 255.0,
                                  green: 203.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)
        let colorWhite = UIColor.white

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorYellow.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorRed.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 3.0 * 2.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorWhite.setFill()
        UIBezierPath.star(at: CGPoint(x: size.width / 2.0, y: (size.height + (size.height / 6.0 * 0.422291236)) / 2.0),
                          radius: size.height / 12.0 * 8.844582472 / 2.0).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
